extensions/resteasy-common/pom.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>quarkus-extensions-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-resteasy-common-spi-parent</artifactId>
+    <name>Quarkus - RESTEasy Common SPI - Parent</name>
+    <description>This module provides reusable abstraction for use with both RESTEasy Classic and Quarkus REST (formerly RESTEasy Reactive)</description>
```

Comment:
To be honest, I would like to avoid such modules. It's better if there is no coupling whatsoever and we just duplicate whatever we need.

Reply:
Hi, I've been looking at this but the issue with duplicating is that the Spring Web extension includes a common core module for both classic and reactive modes, which is responsible for creating the predicate. If we duplicate, it will lead to an error because Quarkus REST and RESTEasy Classic cannot be mixed. For this reason, I don’t see how this approach could work effectively.

Reply:
It can publish, the problem is when consuming. For instance, the tests include both dependencies in the classpath, which is not supported.

Reply:
I'll have to actually run that because I fail to see the problem

Reply:
Thanks I'll check

Reply:
This is likely caused by https://github.com/aureamunoz/quarkus/blob/88c7b352ebf9dccbd5e573fa9e33478c005681ae/extensions/spring-web/core/deployment/pom.xml#L49. 

`quarkus-spring-web-deployment` should only depend on SPI modules from RESTEasy Classic and Reactive

Reply:
Ok, you're right. I managed to put the class in the right place, yes. Thanks. Once you review and validate it I will squash the commits

Reply:
Please go ahead and squash the commits